spec/asyncapi.md

```diff
@@ -144,6 +145,11 @@ connected parts at the discretion of the author. In the latter case, [Reference
 
 By convention, the AsyncAPI Specification (A2S) file is named `asyncapi.json` or `asyncapi.yaml`.
 
+### <a name="relativeReferencesURL"></a>Relative References in URLs
+
+Unless specified otherwise, all properties that are URLs MAY be relative references as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-4.2).
+Unless specified otherwise, relative references are resolved using the URLs defined in the [`Server Object`](#serverObject) as a Base URL. Note that these themselves MAY be relative to the referring document.
```

Comment:
do you mean that if there is URL pointing to license or external docs and it is a relative URL, it should be resolved with Server URL as base URL?

Reply:
@derberg yes, this is how relative references in URLs are handled in OpenAPI 3.1. The reasoning is that assets like external documentation, licenses or authentication needs to live on a server as they are concrete assets. Without server being involved, the only other `baseURI` that relative URLs can be resolved against is the location of the definition document itself, which is in most cases is not what you want.

Reply:
I don't think we can apply the same principle here. Note that in many (most?) cases these servers are not using HTTP but a messaging protocol. How is this supposed to be served then? I understand it makes sense in OpenAPI, where everything is HTTP, but it's not clear to me it works here.

Reply:
@fmvilas good point. You can technically still resolve relative URLs against not HTTP protocol based URLs, but as you pointed out it just might not make sense. But isn't it a responsibility of the definition author so that data within the definition make sense?

I can see two options:

1.) Extending the explanation of `HTTP` protocol

> Unless specified otherwise, relative references are resolved using the HTTP URLs defined in the [`Server Object`](#serverObject) as a Base URL. Note that these themselves MAY be relative to the referring document.

But one can still use use other protocols that browsers can handle like `ftp://` or `file://` or others. That goes to my point of responsibility of definition author.

2.) Provide clarification that only absolute URLs are supported

We will specify what `MUST be in the format of a URL` actually means - URLs must comply to [rfc3986 section 4.3](https://datatracker.ietf.org/doc/html/rfc3986#section-4.3)

Reply:
I'd vote for absolute URLs. License, docs, etc. should always point to production (or final stage) environments IMHO.

Reply:
> I'd vote for absolute URLs.

Well if we agree to specify absolute URLs only, it will clarify the specification and simplify the tooling implementation. So I tend to agree with this as well.

> License, docs, etc. should always point to production (or final stage) environments IMHO.

Yep, if we assume that relative URLs of license, docs, etc.. should successfully resolve against any of the `Server Object` - that will never be true, as many `Server Objects` (as you pointed out) will be defined for protocols other then HTTP or FTP or FILE.

---

I'll redo the PR. And thanks for the input.

Reply:
Thanks for all the work you're doing mate!